src/features/metadata-based-view/MetadataQueryAPIHelper.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * @flow strict
+ * @file Versions Sidebar API Helper
+ * @author Box
+ */
+import find from 'lodash/find';
+import getProp from 'lodash/get';
+import includes from 'lodash/includes';
+import API from '../../api';
+
+import { METADATA_FIELD_TYPE_ENUM, METADATA_FIELD_TYPE_MULTISELECT, ITEM_TYPE_FILE } from '../../common/constants';
+
+import type {
+    MetadataQuery as MetadataQueryType,
+    MetadataQueryResponseData,
+    MetadataQueryResponseEntry,
+    MetadataQueryResponseEntryMetadata,
+    MetadataTemplateSchemaResponse,
+} from '../../common/types/metadataQueries';
+
+type SuccessCallback = (metadataQueryCollection: Collection) => void;
+type ErrorCallback = (e: ElementsXhrError) => void;
+
+const SELECT_TYPES: Array<typeof METADATA_FIELD_TYPE_ENUM | typeof METADATA_FIELD_TYPE_MULTISELECT> = [
+    METADATA_FIELD_TYPE_ENUM,
+    METADATA_FIELD_TYPE_MULTISELECT,
+];
+
+export default class MetadataQueryAPIHelper {
+    api: API;
+
+    metadataQueryResponseData: MetadataQueryResponseData;
```

Comment:
I would still encourage you to keep this file stateless, if possible, and pass the necessary data as parameters to each promise chain handler. You could also use await/async if it helps clean up the chain of method calls.

Reply:
As discussed offline, the promise chain <--> instance variables can not result in race condition as anywhere the promise chain is broken, the errorHandler is called which is not referencing any instance variables.